### PR TITLE
Kenzu: ready to play

### LINKED ISSUE #1792: Enable Kenzu

### PR DESCRIPTION
**Pull request by saugon**
Closes #1792 

> [!WARNING]  
> Needs to be merge alongside https://github.com/lambdaclass/mirra_backend/pull/830

## Motivation
This PR goal is to set Kenzu as a playable character in the game.

_⚠️ Disclaimer 1: This PR was originally intended to be a quick update for both Kenzu and Otix controllers to unblock the backend (hence some Otix updates as well). However, since Otix had some unexpected issues, the PR turned into a ‘ready to play’ implementation for Kenzu. For faster implementation, Otix changes are still included in this PR but are not supposed to be reviewed as ‘ready’._

_⚠️ Disclaimer 2: Some values, such as Kenzu’s speed and ultimate range, will be adjusted in the configurator later._

## Summary of changes
- Added logic to enable movement during skill execution (for the ultimate).
- Implemented combo logic for Kenzu's basic skill (animations and VFX).
- Refactored Kenzu's animation controller.
- Integrated VFX handling for combo animations.
- Fixed Kenzu's material and texture.
- Added new sprites.

To improve in future iterations:
- Dash skill is missing the vfx.

## How has this been tested?
- Playing as Kenzu from start to end.
- Playing with the other characters to verify they work as usual.

## Checklist
- [x] I have tested the changes locally.
- [x] I have tested the whole game after applying the changes, not only the affected areas.
- [ ] I self-reviewed the changes on GitHub, line by line.
- [x] I have tested the changes in another devices.
  - [x] Tested in iOS.
  - [ ] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by alfopisano — COMMENTED** [review]
Nice!! Could play with Kenzu and other characters without any troubles. I just have two comments but maybe they are intended to be adjusted on a future iteration. 
1. I feel Kenzu is much faster than the other characters and is really difficult to escape from him.
2. The animation at the deathsplash is getting stuck very quickly. I know this is a known issue but I feel on this character it happens a lot faster.

**Comment by saugon** [comment]
> Nice!! Could play with Kenzu and other characters without any troubles. I just have two comments but maybe they are intended to be adjusted on a future iteration.
> 
>     1. I feel Kenzu is much faster than the other characters and is really difficult to escape from him.
> 
>     2. The animation at the deathsplash is getting stuck very quickly. I know this is a known issue but I feel on this character it happens a lot faster.

1. Yes, don't worry about that, we have to tune the back configuration. 
2. I agree, is not the ideal but it's not currently a blocker.
3. I’ll add it in the description, but we also have to adjust the ultimate range to match with the VFX at least, so don't worry about that either.

**Review by alfopisano — DISMISSED** [review]
Nice!! Tested on my computer and iOS device playing as Kenzu and other characters and everything worked great

**Review by AngieDutra — COMMENTED** [review]
I couldn't see Kenzu's dash VFX
Kenzu's ulti's VFX isn'r sized to the actual damage it does
His basic VFX starts a bit late and is our of sync with the animation

Other than that I could play with him, he seems to have all his asstes

File: client/Assets/VFX-Tools/PIngle_VFX/VFXGraphs/vfxgraph_H4ck_Denial_AOE.vfx
[the remark this answers is not in the document]
Comment (saugon) [reply client/Assets/VFX-Tools/PIngle_VFX/VFXGraphs/vfxgraph_H4ck_Denial_AOE.vfx:1]: They are auto-updated by Unity, I added them here to solve this that was being super annoying.

**Review by AngieDutra — DISMISSED** [review]
Since my comments will be done in the future this PR is good to go! 
The changes to the VFX were intended
I just created an [issue](https://github.com/lambdaclass/champions_of_mirra/issues/2020) for an unkown bug here but it should not block this to be merged

✅ Tested in Android

**Review by alfopisano — APPROVED** [review]
Nice!! Tested on iOS device and

**Review by AngieDutra — COMMENTED** [review]
Just realized we didn't prewarm kenzu's VFX:
Screenshot 2024-08-08 at 15 24 07
If it's not done in this PR I'll do an issue on this

**Review by AngieDutra — APPROVED** [review]
Nice! 🐱

**Review by AngieDutra — APPROVED** [review]

**Pull request merged**